src/main/scala/io/citrine/lolo/trees/regression/RegressionTrainingLeaf.scala

```diff
@@ -0,0 +1,41 @@
+package io.citrine.lolo.trees.regression
+
+import io.citrine.lolo.trees.{ModelLeaf, ModelNode, TrainingNode}
+import io.citrine.lolo.{Learner, Model, PredictionResult}
+
+import scala.collection.mutable
+
+/**
+  * Created by maxhutch on 3/8/17.
+  */
+class RegressionTrainingLeaf(
+                              trainingData: Seq[(Vector[AnyVal], Double, Double)],
+                              leafLearner: Learner,
+                              depth: Int
+                            ) extends TrainingNode(trainingData, depth) {
+
+  /**
+    * Average the training data
+    *
+    * @return lightweight prediction node
+    */
+  def getNode(): ModelNode[PredictionResult[Double]] = {
+    new ModelLeaf(leafLearner.train(trainingData).getModel().asInstanceOf[Model[PredictionResult[Double]]], depth)
+  }
+
+  def getFeatureImportance(): scala.collection.mutable.ArraySeq[Double] = {
+    importance match {
+      case Some(x) =>
+        val expectations: (Double, Double) = trainingData.map{ case (v, l, w) =>
+          (l * w, l * l * w)
+        }.reduce((u: (Double, Double), v: (Double, Double)) => (u._1 + v._1, u._2 + v._2))
+        val impurity = expectations._2 / trainingData.size - Math.pow(expectations._1 / trainingData.size, 2.0)
```

Comment:
This is not relevant to this particular code review, but why isn't the impurity stored at each node?  Why do we have to recalculate it here, when it was already calculated when the node was split?